ocs_ci/deployment/flexy.py

```diff
@@ -0,0 +1,277 @@
+        # then we don't clone private-conf repo else
+        # we will look for 'flexy_private_conf_url' , if not specified
+        # we will go ahead with default 'flexy-ocs-private' repo
+        if not config.ENV_DATA.get('flexy-env-file'):
+            self.flexy_private_conf_url = config.ENV_DATA.get(
+                'flexy_private_conf_url',
+                constants.FLEXY_DEFAULT_PRIVATE_CONF_URL
+            )
+            # Host dir path for private_conf dir where private-conf will be
+            # cloned
+            self.flexy_host_private_conf_dir_path = os.path.join(
+                self.flexy_host_dir, 'flexy-ocs-private'
+            )
+            self.flexy_env_file = os.path.join(
+                self.flexy_host_private_conf_dir_path, constants.FLEXY_DEFAULT_ENV_FILE
+            )
+        else:
+            self.flexy_env_file = config.ENV_DATA['flexy-env-file']
+
+    def deploy_prereq(self):
+        """
+          Common flexy prerequisites like cloning the private-conf
+          repo locally and updating the contents with user supplied
+          values
+
+          """
+        if not os.path.exists(self.flexy_host_dir):
+            os.mkdir(self.flexy_host_dir)
+            os.chmod(self.flexy_host_dir, mode=0o777)
+        # If user has provided absolute path for env file (through
+        # '--flexy-env-file <path>' CLI option)
+        # then no need of clone else continue with default
+        # private-conf repo and branch
+
+        if not config.ENV_DATA.get('flexy-env-file'):
+            self.clone_and_unlock_ocs_private_conf()
+            config.FLEXY['VARIABLES_LOCATION'] = os.path.join(
+                constants.OPENSHIFT_MISC_BASE, self.template_file
+            )
+            config.FLEXY['INSTANCE_NAME_PREFIX'] = self.cluster_name
+            self.merge_flexy_env()
+
+    def run_container(self, cmd_string):
+        """
+        Actual container run happens here, a thread will be
+        spawned to asynchronously print flexy container logs
+
+        Args:
+            cmd_string (str): Podman command line along with options
+
+        """
+        logger.info(
+            f"Starting Flexy container with options {cmd_string}"
+        )
+        with subprocess.Popen(
+            cmd_string,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            bufsize=1,
+            universal_newlines=True
+        ) as p:
+            for line in p.stdout:
+                logger.info(line)
+            if p.returncode:
+                logger.error("Flexy command failed")
+                raise CalledProcessError(p.returncode, p.args)
+            logger.info("Flexy run finished successfully")
+
+    def build_install_cmd(self):
+        """
+        Build flexy command line for 'deploy' operation
+
+        """
+        cmd = shlex.split("sudo podman run --rm=true")
+        flexy_container_args = self.build_container_args()
```

Comment:
`self.build_container_args()` expects  `purpose` argument (there is no default value)

Reply:
done